filebeat/harvester/registry.go

```diff
@@ -68,13 +72,12 @@ func (r *Registry) Start(h Harvester) {
 	}
 
 	r.wg.Add(1)
-	r.harvesters[h.ID()] = h
-
 	go func() {
 		defer func() {
 			r.remove(h)
 			r.wg.Done()
 		}()
+		r.add(h)
```

Comment:
The only time the prospector interacts with the registry is when adding harvesters or on stopping it. The reason I moved it inside is that previously a harvester was added to the registry and it could happen it was started later and stopped before. With this change it is less likely but probably still possible. That is why I also added the to the harvester so in case it was already stopped, it will not start and directly return. That means in this case the harvester would also not have to be stopped, as it is already finished.